packages/data-context/src/actions/LocalSettingsActions.ts

```diff
@@ -14,19 +14,25 @@ export interface LocalSettingsApiShape {
 export class LocalSettingsActions {
   constructor (private ctx: DataContext) {}
 
-  setPreferences (stringifiedJson: string, type: 'global' | 'project') {
+  async setPreferences (stringifiedJson: string, type: 'global' | 'project') {
     const toJson = JSON.parse(stringifiedJson) as AllowedState
 
-    // update local data
-
-    _.merge(this.ctx.coreData.localSettings.preferences, toJson)
```

Comment:
This line was (harmlessly) in the wrong place because `localSettings.preferences` is exclusively dealing with global state, it was missed in an earlier refactor. Because we were accidentally setting everything here, we were able to write a straightforward unit test to check this `localSettings.preferences` in memory, but it was providing false confidence since this is not the same as what gets written to the files.

I do think we should have a unit test here, but it will take a bit of designing to avoid replacing so much of the function that we are only testing our test code.